makiflow/layers/initializers.py

```diff
@@ -0,0 +1,443 @@
+# Copyright (C) 2020  Igor Kilbas, Danil Gribanov, Artem Mukhin
+#
+# This file is part of MakiFlow.
+#
+# MakiFlow is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# MakiFlow is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Foobar.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+from abc import abstractmethod, ABC
+
+# Some initialize methods
+# Initializations define the way to set the initial random weights of MakiFlow layers.
+
+# (in_f, out_f), in most cases
+NUM_OF_PARAMS_DENSE = 2
+
+
+class BaseInitializer(ABC):
+    Name = 'Base'
+
+    def __str__(self):
+        return self.Name
+
+    @abstractmethod
+    def __call__(self, shape: list, dtype=np.float32):
+        pass
+
+    def _check_input_shape(self, shape: list) -> list:
```

Comment:
This method (as well as random tensor generation) is being called in every __call__ of all child classes. Try to get rid of this repetition by calling it somewhere else in the parent class.
I would suggest to create two additional parent classes (gauss and uniform) that generate purely random tensors:
```
class Gaussian(BaseInitializer):
    def __call__(shape, dtype):
        shape = self._check_input_shape(shape)
        w = np.random.randn(*shape).astype(dtype)
        return w

class He(Gaussian):
    def __call__(self, shape):
        w = super().__call__(shape)
        w *= np.sqrt(2 / np.prod(shape[:-1]))
        return w
```